server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStore.java

```diff
@@ -213,10 +225,10 @@ else if (map.get(USER_AUTHENTICATION_UAA_PRINCIPAL)!=null) {
     protected void performExpirationClean() {
         Instant last = lastClean.get();
         //check if we should expire again
-        Instant now = Instant.now();
+        Instant now = Instant.now(clock);
         if (enoughTimeHasPassedSinceLastExpirationClean(last, now)) {
             //avoid concurrent deletes from the same UAA - performance improvement
-            if (lastClean.compareAndSet(last, last.plus(getExpirationTime()))) {
+            if (lastClean.compareAndSet(last, now)) {
```

Comment:
to me this is the only line which need to be fixed, so compareAndSet needs the correct new value which is now

Reply:
> to me this is the only line which need to be fixed

I agree, now especially as the other alternate PR does not have sleep() in the unit test code anymore.